### PR TITLE
feat(earn): Allow claiming a delayed withdraw

### PR DESCRIPTION
**Pull request by th0rOdinson**

### CODE REVIEW

File: packages/ui-library/src/components/transaction-receipt/index.tsx [hunk 1]
```diff
@@ -619,26 +620,78 @@ const EarnWithdrawTransactionReceipt = ({ transaction }: { transaction: EarnWith
   const { spacing } = useTheme();
   const intl = useIntl();
 
+  return (
+    <>
+      <StyledSectionContent>
+        <Typography variant="labelRegular">
+          <FormattedMessage description="TransactionReceipt-transactionEarnWithdraw" defaultMessage="Withdrew" />
```
Comment (FiboApe) [inline packages/ui-library/src/components/transaction-receipt/index.tsx:627]: wait are we parsing all sending of delayed withdraw and normal withdraws with the same receipt? Or does this only apply if you also withdrew some token through market?

File: apps/root/src/common/utils/transaction-history/parsing.tsx [hunk 2]
```diff
@@ -91,15 +93,14 @@ const parseDcaCreatedApiEvent: ParseFunction<DCACreatedApiEvent, DCACreatedEvent
...
-        amountInUSD:
-          event.data.fromToken.price === null || isUndefined(event.data.fromToken.price)
-            ? undefined
-            : parseFloat(
-                formatUnits(
-                  BigInt(event.data.rate) * parseNumberUsdPriceToBigInt(event.data.fromToken.price),
-                  dcaBaseEventData.fromToken.decimals + 18
-                )
-              ).toFixed(2),
+        amountInUSD: isNil(event.data.fromToken.price)
```
Comment (FiboApe) [inline apps/root/src/common/utils/transaction-history/parsing.tsx:96]: why was this change needed? isNil is going to return false if it is undefined as well

File: apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/delayed-withdraw-container/index.tsx [hunk 3]
```diff
@@ -124,14 +130,31 @@ const DelayedWithdrawItem = ({ delayed, type, position }: DelayedWithdrawItemPro
...
-      {type === DelayedWithdrawalStatus.READY && (
-        <Button variant="outlined">
-          <FormattedMessage
-            description="earn.strategy-management.withdraw.delayed-withdraw.item.withdraw-now"
-            defaultMessage="Withdraw now"
-          />
-        </Button>
-      )}
+      {type === DelayedWithdrawalStatus.READY &&
+        (!!position.pendingTransaction ? (
```
Comment (FiboApe) [inline apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/delayed-withdraw-container/index.tsx:134]: No need for this since they can submit several delayed withdraws at the same time

File: apps/root/src/hooks/earn/useEarnClaimDelayedWithdrawAction.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      try {
+        if (!claimToken) throw new Error('No claimable token found');
+
+        setModalLoading({
+          content: (
+            <Typography variant="bodyRegular">
+              <FormattedMessage
+                description="earn.strategy-management.claim-delayed-withdraw.modal.loading"
+                defaultMessage="You are now claiming delayed withdraw from {farm}. Time to enjoy the rewards you've cultivated 🎉"
+                values={{ farm: strategy.farm.name }}
```
Comment (FiboApe) [inline apps/root/src/hooks/earn/useEarnClaimDelayedWithdrawAction.tsx:35]: lets also add the token symbol here and the amount in case the copy team want to use it

File: apps/root/src/services/earnService.ts [hunk 5]
```diff
@@ -681,6 +681,56 @@ export class EarnService extends EventsManager<EarnServiceData> {
...
+      throw new Error('Could not find userStrategy');
+    }
+
+    const strategy = this.allStrategies.find((s) => s.id === userStrategy.strategy);
+
+    if (!strategy) {
+      throw new Error('Could not find strategy');
+    }
+
+    // NOTE: For the moment we should never expect the permissionSignature.
```
Comment (FiboApe) [inline apps/root/src/services/earnService.ts:705]: Like it that we already support it tho :D

File: packages/ui-library/src/components/transaction-receipt/index.tsx (hunk 1 above)
Comment (th0rOdinson) [reply packages/ui-library/src/components/transaction-receipt/index.tsx:627]: Yes, they are all the same event, with a different withdrawType prop. I've only added the withdraw Type in the delayed receipt, becuase it's the one that doesn't put money immediately.
Should we have three different receipts?

File: apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/delayed-withdraw-container/index.tsx (hunk 3 above)
Comment (th0rOdinson) [reply apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/delayed-withdraw-container/index.tsx:134]: This would only disable that particular button, so they don't submit the same transaction, and they can check status

Comment (FiboApe) [reply apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/delayed-withdraw-container/index.tsx:134]: Ah right since we have only 1 pending and 1 ready by position

File: packages/ui-library/src/components/transaction-receipt/index.tsx [hunk 6]
```diff
@@ -620,47 +620,40 @@ const EarnWithdrawTransactionReceipt = ({ transaction }: { transaction: EarnWith
   const { spacing } = useTheme();
   const intl = useIntl();
 
+  const isDelayed = transaction.data.withdrawn.some(
```
Comment (FiboApe) [inline packages/ui-library/src/components/transaction-receipt/index.tsx:623]: can we move this to a util function? already seen it used 3 times

**Review by FiboApe — APPROVED** [review]
